buildscripts/docker/Dockerfile.windows_builder

```diff
@@ -0,0 +1,236 @@
+COPY build_toxcore_windows.sh /build/build_toxcore_windows.sh
+RUN mkdir -p /src/tox && \
+  cd /src/tox && \
+  /build/build_toxcore_windows.sh && \
+  rm -fr /src/tox
+
+RUN mkdir /export && \
+  cp /usr/${ARCH}-w64-mingw32/lib/libwinpthread-1.dll /export/ && \
+  cp /usr/lib/gcc/${ARCH}-w64-mingw32/10-posix/libgcc_s_*-1.dll /export && \
+  cp /usr/lib/gcc/${ARCH}-w64-mingw32/10-posix/libstdc++-6.dll /export && \
+  cp /usr/lib/gcc/${ARCH}-w64-mingw32/10-posix/libssp-0.dll /export && \
+  cp /windows/bin/Qt5Core.dll /export && \
+  cp /windows/bin/Qt5Gui.dll /export && \
+  cp /windows/bin/Qt5Network.dll /export && \
+  cp /windows/bin/Qt5Svg.dll /export && \
+  cp /windows/bin/Qt5Xml.dll /export && \
+  cp /windows/bin/Qt5Widgets.dll /export && \
+  cp /windows/bin/avcodec-*.dll /export && \
+  cp /windows/bin/avdevice-*.dll /export && \
+  cp /windows/bin/avformat-*.dll /export && \
+  cp /windows/bin/avutil-*.dll /export && \
+  cp /windows/bin/libexif-*.dll /export && \
+  cp /windows/bin/libqrencode.dll /export && \
+  cp /windows/bin/libsodium-*.dll /export && \
+  cp /windows/bin/libsqlcipher-*.dll /export && \
+  cp /windows/bin/swscale-*.dll /export && \
+  cp /windows/bin/libcrypto-*.dll /export && \
+  cp /windows/bin/libtoxcore.dll /export && \
+  cp /windows/bin/libopus-*.dll /export && \
+  cp /windows/lib/libvpx.dll /export && \
+  cp /windows/bin/libssl-*.dll /export && \
+  cp /windows/bin/libsnore-qt5.dll /export && \
+  mkdir -p /export/libsnore-qt5/ && \
+  cp /windows/plugins/libsnore-qt5/libsnore_backend_windowstoast.dll /export/libsnore-qt5/ && \
+  cp /windows/bin/SnoreToast.exe /export && \
+  cp -r /windows/plugins/iconengines /export && \
+  cp -r /windows/plugins/imageformats /export && \
+  cp -r /windows/plugins/platforms /export
+
+RUN mkdir -p /debug_export
+
+COPY download/download_mingw_debug_scripts.sh /build/download/download_mingw_debug_scripts.sh
+RUN mkdir -p /src/mingw-debug-scripts && \
+  cd /src/mingw-debug-scripts && \
+  /build/download/download_mingw_debug_scripts.sh  && \
+  sed -i "s|your-app-name.exe|qtox.exe|g" debug-*.bat && \
+  cp -a debug-*.bat /debug_export && \
+  rm -fr /src/mingw-debug-scripts
+
+COPY download/download_gmp.sh /build/download/download_gmp.sh
+COPY build_gmp_windows.sh /build/build_gmp_windows.sh
+RUN  mkdir -p /src/gmp && \
+  cd /src/gmp && \
+  /build/build_gmp_windows.sh && \
+  rm -fr /src/gmp
+
+COPY download/download_libexpat.sh /build/download/download_libexpat.sh
+COPY build_libexpat_windows.sh /build/build_libexpat_windows.sh
+RUN mkdir -p /src/libexpat && \
+  cd /src/libexpat && \
+  /build/build_libexpat_windows.sh && \
+  rm -fr /src/libexpat
+
+COPY download/download_gdb.sh /build/download/download_gdb.sh
+COPY build_gdb_windows.sh /build/build_gdb_windows.sh
+RUN mkdir -p /src/gdb && \
+  cd /src/gdb && \
+  /build/build_gdb_windows.sh && \
+  rm -fr /src/gdb && \
+  cp /windows/bin/gdb.exe /debug_export/gdb.exe
+
+RUN mkdir -p /qtox
+WORKDIR /qtox
+
```

Comment:
The way the Windows build worked, is that it cached all the deps separately into a provided docker volume. This allowed you to remove a dependency directory, say ffmpeg, and have the build script rebuild only that dependency and qTox. It didn't have to rebuild any dependency before ffmpeg and also any dependency after ffmpeg. However, with how Docker layers work, it will have to rebuild everything after ffmpeg, which is a degradation.

Reply:
I partially agree, partially disagree here. Yes it's a regression in that you cannot force recompile a single dependency if it's not the last in the chain, however it is now impossible to rebuild a dependency of a dep without rebuilding it's dependents.

I think the build of everything in this image is relatively cheap other than Qt/OpenSSL. I was willing to take the hit there as dependency updates are infrequent and the cost is low in most cases.

This is fixable by using separate dep building docker images and FROM statements, I just don't think it's worth the additional complexity